modules/gltf/gltf_document.cpp

```diff
@@ -1539,7 +1544,7 @@ GLTFAccessorIndex GLTFDocument::_encode_accessor_as_ints(Ref<GLTFState> p_state,
 	GLTFBufferIndex buffer_view_i;
 	int64_t size = p_state->buffers[0].size();
 	const GLTFType type = GLTFType::TYPE_SCALAR;
-	const int component_type = GLTFDocument::COMPONENT_TYPE_INT;
+	const int component_type = max_index > 65535 || p_for_vertex ? GLTFDocument::COMPONENT_TYPE_INT : GLTFDocument::COMPONENT_TYPE_UNSIGNED_SHORT;
```

Comment:
```suggestion
	int component_type;
if (max_index > 65535 || p_for_vertex) {
    component_type = GLTFDocument::COMPONENT_TYPE_INT;
} else {
    component_type = GLTFDocument::COMPONENT_TYPE_UNSIGNED_SHORT;
}
```

Might want to unpack the C++ idiom so that it's easier to review.

Reply:
This is good that we switch from int to short if the count is good. We probably shouldn't switch to 8 bit (256 values) because I'm worried about compatibility. Not sure.